### ISSUE TITLE
WeightedDnsElasticLoadBalancer doesn't remove MainDomain

### ISSUE THREAD

**Issue by lukasniemeier-zalando** [zalando-stups/senza#184]
It seems like if you make changes to the `MainDomain` resources of a `WeightedDnsElasticLoadBalancer` it will not get deleted on `senza delete`.

1. Create a stack with a `WeightedDnsElasticLoadBalancer`
2. Shift some traffic to your stack with `senza traffic` (or by changing the weight of the DNS record in the AWS console)
3. Delete the stack with `senza del`

CF will show an event that the `MainDomain` record was removed successfully, but Route53 will still show the record after deletion of the stack has finished.

Since deletion of the record works if you _reset_ the weight of the record back to zero (as it is specified in the CF template for this record) it looks like CF removes only those resources which fit all parameters as specified in the template. It's still strange the CF reports the removal as a success.

Do we have to update resources of a stack (e.g. changing DNS weights) via CF `update_stack`?

**Comment by i8r** [comment]
I ran into the same issue. Additionally there is the problem if you try to deploy a stack of the same application with `senza create` (after you deleted a stack like described). In this case the stack creation fails because the domain already exists.

**Comment by hjacobs** [comment]
@lukasniemeier-zalando @i8r copying my comment from #268:

I think this behavior is due to how CloudFormation works:

* Senza `create` creates a new CloudFormation (CF) stack with VersionDomain and MainDomain DNS records
  * MainDomain has weight zero for this stack version and is a "managed by CF" resource
* Senza `traffic` will change the weight of the MainDomain record without changing the CF template
  * MainDomain is therefore no longer exactly the same resource as originally created by CF
* Senza `delete` will delete the CF stack
  * MainDomain will not be deleted by CF as the resource no longer matches the original "CF managed" resource

So basically we are mixing CloudFormation and manual changes (directly using Route 53 API), i.e. the manually changed MainDomain DNS record leaves the scope of CF management during the lifetime.

You can observe that a brand new stack will be completely deleted by CF if you don't use `senza traffic` during its lifetime.

**Comment by lukasniemeier-zalando** [comment]
> So basically we are mixing CloudFormation and manual changes (directly using Route 53 API), i.e. the manually changed MainDomain DNS record leaves the scope of CF management during the lifetime.

Correct see my comment above, senza should do everything with `update_stack` of the Cloudformation API.
